phac-nml/irida-next
# Namespace Group Link UI - Listing

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds in a `Groups` tab on the members page which lists the namespace group links. The members index was also refactored into partials to support this new view and to simplify turbo stream responses.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

`Members page with tabs`

![image](https://github.com/phac-nml/irida-next/assets/25466211/a8107182-3cc8-4337-9cc8-0b6e0715c80b)

`Group -> Members -> Groups tab`

![image](https://github.com/phac-nml/irida-next/assets/25466211/d8ad4d1c-e1d9-4b08-a439-d5fa63cb3621)

`Project -> Members -> Groups tab`

![image](https://github.com/phac-nml/irida-next/assets/25466211/f3860790-0318-49b0-a578-2a3927bd92d6)


## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Login as admin@email.com
2. Visit any group 
3. Go to it's members page
4. You should see 2 tabs (Members and Groups)
5. Click the Groups tab
6. Verify the namespace group links are listed and the membership source is correct
7. Now try the same thing with a project
8. Verify the namespace group links are listed and the membership source is correct 

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
